docs/specification.md

```diff
@@ -21,6 +21,19 @@ These specifications were assembled with the following references:
 
 - The token contract must store tokens of a single type.
 
+# Questions
+
+1. How many addresses can be assigned to each role?
+We have made some assumptions, see the [Roles](#roles) section below, please check them.
+For non-unique roles: is there an upper limit on the number of addresses?
+2. Can one address have more than one role?
```

Comment:
I don't think so due to the problems I mentioned earlier, but who knows what guys from TQ have in mind